src/prefect/_internal/compatibility/migration.py

```diff
@@ -34,6 +34,7 @@
 
 REMOVED_IN_V3 = {
     "prefect.client.schemas.objects:MinimalDeploymentSchedule": "Use `prefect.client.schemas.actions.DeploymentScheduleCreate` instead.",
+    "prefect.context.PrefectObjectRegistry": ugrade_guide_msg,
```

Comment:
I don't think `PrefectObjectRegistry` is a user facing object, we never encouraged its direct use and anecdotally I've never seen anyone using it directly.

I think the items in `REMOVED_IN_V3` should be user facing objects that we've removed

Reply:
I was just working on upgrading the integration template (collection template) and tried to make a test integration and it was in the tests and broke because it doesn't exist in V3. 

Other folks might be using it in their tests if they followed our lead there. And anyone with a 2.x integration built on the template will be facing this if upgrading to 3.x. So I feel like it's user-facing enough. Also, I don't think there's any harm in adding it to the list - you only encounter it if you use it.